scilpy/tracking/seed.py

```diff
@@ -264,3 +265,104 @@ def init_generator(self, rng_seed, numbers_to_skip):
+        self.space = Space.VOXMM
+        self.origin = Origin.NIFTI
+
+        self.centerlines = centerlines
+        self.diameters = diameters
+        self.nb_seeds_per_fibertube = nb_seeds_per_fibertube
+
+    def init_generator(self, rng_seed, numbers_to_skip):
+        """
+        Initialize a numpy number generator according to user's parameters.
+        Returns also the shuffled index of all fibertubes.
+
+        The values are not stored in this classed, but are returned to the
+        user, who should add them as arguments in the methods
+        self.get_next_pos()
+        self.get_next_n_pos()
+        The use of this is that with multiprocessing, each process may have its
+        own generator, with less risk of using the wrong one when they are
+        managed by the user.
+
+        Parameters
+        ----------
+        rng_seed : int
+            The "seed" for the random generator.
+        numbers_to_skip : int
+            Number of seeds (i.e. voxels) to skip. Useful if you want to
+            continue sampling from the same generator as in a first experiment
+            (with a fixed rng_seed).
+
+        Return
+        ------
+        random_generator : numpy random generator
+            Initialized numpy number generator.
+        indices : ndarray
+            Shuffled indices of current seeding map, shuffled with current
+            generator.
+        """
+        self.generator = np.random.RandomState(rng_seed)
+
+        # 1. Initializing seeding maps indices (shuffling in-place)
+        indices = np.arange(len(self.centerlines))
+        self.generator.shuffle(indices)
+
+        # 2. Generating the seed for the random sampling.
+        # Because FibertubeSeedGenerator uses rejection sampling to seed
+        # within a cylinder, we can't predict how many generator calls will
+        # be done in each thread to avoid duplicates. We instead generate a
+        # single, predictable number used as a seed for the rejection
+        # sampling.
+        while numbers_to_skip > 100000:
+            self.generator.random_sample(100000)
+            numbers_to_skip -= 100000
+        self.generator.random_sample(numbers_to_skip)
+        sampling_rng_seed = self.generator.randint(0, 2**32-1)
+        self.sampling_generator = np.random.default_rng(sampling_rng_seed)
+
+        return self.sampling_generator, indices
+
+    def get_next_pos(self, random_generator: np.random.Generator,
+                     shuffled_indices, which_seed):
+
+        which_fi = which_seed // self.nb_seeds_per_fibertube
+
+        fiber = self.centerlines[shuffled_indices[which_fi]]
+        radius = self.diameters[shuffled_indices[which_fi]] / 2
+
+        seed = sample_cylinder(fiber[0], fiber[1], radius, 1,
+                               random_generator)[0]
+
+        return seed[0], seed[1], seed[2]
+
+    def get_next_n_pos(self, random_generator, shuffled_indices,
+                       which_seed_start, n):
```

Comment:
docstring ?

Reply:
This class inherits SeedGenerator and these functions don't change the original interface/docstring. It works well and shows up when mouse hovering. Should I still duplicate the original docstring?